docs/howto/writing-an-application-from-scratch/index.adoc

```diff
@@ -96,6 +96,7 @@ myApp
 [discrete]
 == Next steps
 
-ifdef::product[* xref:deploying-an-application-to-openshift[Deploy the application to Openshift]]
-* Package the application in an xref:creating-an-uberjar[uberjar].
+* Package the application in an uberjar.
+ifdef::product[* Deploy the application to Openshift.]
```

Comment:
Why links no good?

Reply:
I wanted to get rid of hard dependencies. I believe the links didn't add that much value, so I removed them. I can undo that change if we want to keep them.

Reply:
Maybe AsciiDoc editors should grow a "Refactor: Rename" functionality :-)

I mean, links are the best thing on the web. Also not sure if it's understandable that each item is still a pointer to some other chapter, if it's not a link.

Reply:
Editing is not really the issue, it's the publication. This module may be used in multiple assemblies where deployment to OpenShift is documented (or it isn't), so the expectation is that the reader will read the assembly from top to bottom.

We are also judicious when using links because we have data that indicate that if the reader sees a link, they click it, and tend to get lost in the related material, so this sections is really just a couple of pointers on what to do next if it's not obvious. I could also remove it altogether.

Reply:
> the expectation is that the reader will read the assembly from top to bottom

Whoa! That is a dangerous assumption to make. I'm going to claim that if the assembly doesn't contain the linked chapter, then the pointer to it shouldn't be present, no matter if it's a link or not.

I'd probably be fine with removing most of the "related info" content, except that all the places that talk about hollow JARs must link to the hollow JAR limitations, that would be a hard constraint from my POV.

Reply:
It's not dangerous, it's necessary as it's user stories (i. e. a task from beginning to end) we want to document ultimately. Right now, the book is laid out like the typical "guide" with all sorts info just cobbled together, but that's not what we want to do in the future.

I will then remove the _Next steps_ subsection as we don't have the assemblies yet.

Reply:
I thought the current layout of a "procedure" comes from a common template, "related info" being part of that. Sounds like the common template should be fixed, too?

Reply:
_Related info_ (or _Additional resources_) is indeed part of the procedure template, but it is optional.